Userland/Services/SystemServer/Service.cpp

```diff
@@ -209,9 +209,9 @@ void Service::spawn(int socket_fd)
             setenv("SOCKET_TAKEOVER", builder.to_string().characters(), true);
         }
 
-        if (m_account.has_value()) {
+        if (m_account.has_value() && m_account.value().uid() != getuid()) {
             auto& account = m_account.value();
-            if (setgid(account.gid()) < 0 || setgroups(account.extra_gids().size(), account.extra_gids().data()) < 0 || setuid(account.uid()) < 0) {
+            if (!account.login()) {
```

Comment:
This {c,sh}ould be a separate commit :)

Reply:
I'm removing it as I plan to modify this again when #14296 will land.